apps/web/src/components/ShareDialog.tsx

```diff
@@ -0,0 +1,97 @@
+// ShareDialog.tsx
+import React, { useState, useEffect } from "react";
+import * as Dialog from "@radix-ui/react-dialog";
+import * as Switch from "@radix-ui/react-switch";
+import { FiShare } from "react-icons/fi";
+import { BiLink } from "react-icons/bi";
+import { AiOutlineCheck } from "react-icons/ai";
+import { TwitterIcon, TwitterShareButton } from "react-share";
+import { ShareToLens, Theme, Size } from "@lens-protocol/widgets-react";
+export type ShareDialogProps = {
+  title: string;
+  content: string;
+};
+
+export const ShareDialog: React.FC<ShareDialogProps> = ({ title, content }) => {
+  const [uniqueURL, setUniqueURL] = useState("");
+  const [pageTitle, setPageTitle] = useState("");
+  const [pageContent, setPageContent] = useState("");
```

Comment:
❓ Do we need this state or can we render this component only when we have the page in the parent component?

```tsx
{page && <ShareDialog {...} />}
```